source/Posets/InductiveTypesfromSmallGeneratedLattice.lagda

```diff
@@ -0,0 +1,1758 @@
+   where
+    P : B → 𝓥  ̇
+    P b = b ≤ᴮ x
+    Q : B → 𝓦  ̇
+    Q b = ((q b) ≤ x) holds
+    e : (b : B) →  b ≤ᴮ x ≃ ((q b) ≤ x) holds
+    e b = _≤ᴮ_-≃-_≤_ {b} {x}
+
+  ↓ᴮ-is-small : {x : ⟨ L ⟩} → ↓ᴮ x is 𝓥 small
+  ↓ᴮ-is-small {x} = (small-↓ᴮ x , small-↓ᴮ-≃-↓ᴮ {x})
+
+  is-sup'ᴮ : (x : ⟨ L ⟩) → x ＝ ⋁ (small-↓ᴮ x , small-↓ᴮ-inclusion x)
+  is-sup'ᴮ x = ≃-families-＝-sup
+               x
+               (⋁ (small-↓ᴮ x , small-↓ᴮ-inclusion x))
+               (is-sup x)
+               (is-lub-for L ((small-↓ᴮ x , small-↓ᴮ-inclusion x)))
+   where
+    open Equivalent-Families-have-same-Join L (↓ᴮ x)
+                                              (small-↓ᴮ x)
+                                              small-↓ᴮ-≃-↓ᴮ
+                                              (↓ᴮ-inclusion x)
+                                              hiding (⋁_)
+
+  is-supᴮ : (x : ⟨ L ⟩)
+          → (x is-lub-of (small-↓ᴮ x , small-↓ᴮ-inclusion x)) holds
+  is-supᴮ x =
+    transport (λ z → (z is-lub-of (small-↓ᴮ x , small-↓ᴮ-inclusion x)) holds)
+              (is-sup'ᴮ x ⁻¹)
+              (is-lub-for L ((small-↓ᴮ x , small-↓ᴮ-inclusion x)))
+
+  is-upper-boundᴮ : (x : ⟨ L ⟩)
+                  → (x is-an-upper-bound-of
+                       (small-↓ᴮ x , small-↓ᴮ-inclusion x)) holds
+  is-upper-boundᴮ x = pr₁ (is-supᴮ x)
+
+  is-least-upper-boundᴮ : (x : ⟨ L ⟩)
+                        → ((u' , _) : upper-bound
+                                      (small-↓ᴮ x , small-↓ᴮ-inclusion x))
+                        → (x ≤ u') holds
+  is-least-upper-boundᴮ x = pr₂ (is-supᴮ x)
+
+\end{code}
+
+Now we construct the least closed subset of an inductive definition as a QIT.
+Since HIT's are not native in Agda we postulate the existence of such a type
+as well as it's induction principle and work with it axiomatically. It is
+conveinient to package this postulate as a Record.
+
+Notice that the QIT has to constructors which representing the closure
+conditions we wish to impose on subsets. The c-cl constructor says that for
+any subset contained in the least closed subset elements in the downset of
+it's join are in the least closed subset. The ϕ-cl constructor says that for
+any a : L and b : B with (b , a) ∈ ϕ and ↓ᴮ a 'contained' in the least
+closed subset then b is in the least closed subset.
+
+We also derive the initiality of the least closed subset from the postulated
+induction principle. Initiality is closely related with the 'least' part of
+our least fixed point theorem.
+
+\begin{code}
+
+module Inductive-Definitions {𝓤 𝓦 𝓥 : Universe}
+                             {B : 𝓥  ̇}
+                             (L : Sup-Lattice 𝓤 𝓦 𝓥)
+                             (q : B → ⟨ L ⟩)
+                              where
+
+ open Small-Basis L q
+ open Joins _≤_
+
+ module Ind-from-Small-Basis-Facts (h : is-small-basis) where
+
+  open Small-Basis-Facts h
+
+  record Inductively-Generated-Subset-Exists (ϕ : 𝓟 {𝓤 ⊔ 𝓥} (B × ⟨ L ⟩)): 𝓤ω
```

Comment:
And a short sentence here saying something like "We now define the QIT discussed above".

Also, use lowercase names for this (like we do e.g. in the `general-set-quotients-exist` or `propositional-truncations-exist`).